task/control-plane-agent/src/inventory/our_inventory.rs

```diff
@@ -0,0 +1,111 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Submodule that defines logical components we claim to have (as opposed to
+//! components entirely managed by other tasks, like I2C devices).
+
+use gateway_messages::{
+    sp_impl::DeviceDescription, DeviceCapabilities, DevicePresence, SpComponent,
+};
+
+pub(super) struct OurInventory {}
+
+impl OurInventory {
+    pub(super) fn new() -> Self {
+        let () = ASSERT_EACH_DEVICE_FITS_IN_ONE_PACKET;
+        Self {}
+    }
+
+    pub(super) fn num_devices(&self) -> usize {
+        DEVICES.len()
+    }
+
+    pub(super) fn device_description(
+        &self,
+        index: usize,
+    ) -> DeviceDescription<'static> {
+        DEVICES[index]
+    }
+}
+
+// List of logical or high-level components that this task is responsible for
+// (or at least responds to in terms of MGS requests for status / update, even
+// if another task is actually responsible for lower-level details).
+//
+// TODO: Are our device names and descriptions good enough, or are there more
+//       specific names we should use? This may be answered when we expand
+//       DeviceDescription with any VPD / serial numbers.
+const DEVICES: &'static [DeviceDescription<'static>] = &[
```

Comment:
We don't need `'static` here.  Oddly, we _do_ need it above (in `device_description`), despite `DeviceDescription` not having an associated lifetime.

Reply:
We can remove the `'static` here because the [elided lifetimes in paths](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#elided-lifetimes-in-paths) lint is "allowed by default". I don't like elided lifetimes in paths, so I made it explicit here. :)

Reply:
Okay, I'm fine keeping it here!